### DISCUSSION TITLE
Deferred with Guzzle PromiseInterface

### DISCUSSION THREAD

**Discussion by vasily-kartashov** [webonyx/graphql-php#1511] [General]
I cannot fully grasp the following use case. A node, which we'll call `User::status`, is resolved by sending a Guzzle request to an HTTP microservice. This microservice expects a list of `userIds` as input. Therefore, the most efficient solution appears to be returning a `Deferred` that resolves to a `PromiseInterface`. The `Deferred` would accumulate `userIds`, and once all `userIds` have been collected, it would send a request to the HTTP microservice, immediately returning a promise. The GraphQL library should then be intelligent enough to recognize that this is a `Promise`/`Thenable` result, and proceed to resolve other nodes before returning to wait for the promise's resolution.

I am using vanilla PHP (without AMP, Swoole, or similar technologies). Is the above scenario feasible? How can I implement it?

**Comment by mfn** [comment]
As you mentioned, wrapping it in a `\GraphQL\Deferred` should to your trick; I use it a lot (not for HTTP necessarily, but dataloaders to resolve DB relations).

Maybe you miss resolving your inner promise in the `Deferred` callback? Can you show some code?
